invoke_test.go

```diff
@@ -19,6 +19,19 @@ var (
 	}
 )
 
+func TestInvokeWithContextTimeout(t *testing.T) {
```

Comment:
Do you want a test where the incoming context has no deadline, to cover that case?